pennylane/devices/qubit/apply_operation.py

```diff
@@ -284,6 +284,9 @@ def apply_mid_measure(
         is_state_batched (bool): Boolean representing whether the state is batched or not
         debugger (_Debugger): The debugger to use
         mid_measurements (dict, None): Mid-circuit measurement dictionary mutated to record the sampled value
+        postselect_mode (str): Configuration for handling shots with mid-circuit measurement
```

Comment:
Mind adding this new keyword argument to the `apply_operation` base function? 

That's where the documentation for `apply_operation` comes from, and I think it's important we keep a well-specified list of all the keyword arguments that can be provided to `apply_operation`. Definitely growing a lot more than I expected it too...